### PR TITLE
TOOLS/INFO: Introduce process placement to control type of selected devices

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Introduce process placement to control type of selected devices.

## Why ?

The PR fixes 1st item of #6271 (comment)
Remote peer placement could be controlled by setting `-P` parameter `{ self, intra, inter }` to emulate a specific transport selection for their communications.

## How ?

1. Introduce `-P` parameter.
2. Select devices based on `-D` and `-P` parameters.

### CODE REVIEW

**Comment by yosefe** [comment]
@dmitrygx AFAIK we discussed that process placement parameter will affect machine guid/reachability tests. otherwise, how is it different that what we can do with -D parameter?

**Comment by dmitrygx** [comment]
> @dmitrygx AFAIK we discussed that process placement parameter will affect machine guid/reachability tests. otherwise, how is it different that what we can do with -D parameter?

@yosefe fixed
@brminich @hoopoepg could you review pls?

**Comment by dmitrygx** [comment]
@brminich @hoopoepg ping

**Review by hoopoepg — APPROVED** [review]

**Comment by dmitrygx** [comment]
@yosefe could you review pls when you have time?

File: src/tools/info/proto_info.c [hunk 1]
```diff
@@ -118,6 +119,9 @@ wait_completion(ucp_worker_h worker, ucs_status_ptr_t status_ptr)
     } else if (UCS_PTR_IS_PTR(status_ptr)) {
         do {
             ucp_worker_progress(worker);
+            if (peer_worker != worker) {
```
Comment (yosefe) [inline src/tools/info/proto_info.c:122]: IMO can remove this if for simplicity

File: src/tools/info/ucx_info.h [hunk 2]
```diff
@@ -27,6 +27,13 @@ enum {
 };
 
 
+typedef enum {
+    PROCESS_SELF,
+    PROCESS_INTRA,
+    PROCESS_INTER
+} process_place_t;
```
Comment (yosefe) [inline src/tools/info/ucx_info.h:34]: process_placement_t
PROCESS_PLACEMENT_SELF
PROCESS_PLACEMENT_INTRA
PROCESS_PLACEMENT_INTER

File: src/tools/info/ucx_info.h [hunk 3]
```diff
@@ -42,7 +49,7 @@ ucs_status_t
 print_ucp_info(int print_opts, ucs_config_print_flags_t print_flags,
                uint64_t ctx_features, const ucp_ep_params_t *base_ep_params,
                size_t estimated_num_eps, size_t estimated_num_ppn,
-               unsigned dev_type_bitmap, const char *mem_size,
-               const char *ip_addr);
+               unsigned dev_type_bitmap, process_place_t proc_place,
```
Comment (yosefe) [inline src/tools/info/ucx_info.h:52]: process_placement_t proc_placement,
or
process_placement_t placement

File: src/tools/info/proto_info.c (hunk 1 above)
Comment (dmitrygx) [reply src/tools/info/proto_info.c]: done

File: src/tools/info/ucx_info.h (hunk 2 above)
Comment (dmitrygx) [reply src/tools/info/ucx_info.h]: fixed

File: src/tools/info/ucx_info.h (hunk 3 above)
Comment (dmitrygx) [reply src/tools/info/ucx_info.h]: fixed

**Review by yosefe — APPROVED** [review]

**Pull request merged**